tests/ops/qubit/test_arithmetic_ops.py

```diff
@@ -200,4 +232,24 @@ def f(state):
             return qml.probs(wires=range(3))
 
         u = np.array([f(state) for state in np.eye(2 ** 3)]).T
-        assert np.allclose(u, qml.QubitSum._matrix())
+        assert np.allclose(u, qml.QubitSum.compute_matrix())
+
+    def test_matrix_representation(self, tol):
+        """Test that the canonical matrix is defined correctly"""
+
+        res_static = qml.QubitSum.compute_matrix()
+        res_dynamic = qml.QubitSum(wires=[0, 1, 2]).matrix()
+        expected = np.array(
+            [
+                [1, 0, 0, 0, 0, 0, 0, 0],
+                [0, 1, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 1, 0, 0, 0, 0],
+                [0, 0, 1, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 1, 0, 0],
+                [0, 0, 0, 0, 1, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 1, 0],
+                [0, 0, 0, 0, 0, 0, 0, 1],
+            ]
+        )
+        assert np.allclose(res_static, expected, atol=tol)
+        assert np.allclose(res_dynamic, expected, atol=tol)
```

Comment:
I'm sorry I test both methods in one here, but I just couldn't get myself to duplicate every test. If needed, we can do that later?